src/Microsoft.VisualStudio.Threading.Analyzers/CommonInterest.cs

```diff
@@ -37,6 +42,82 @@ internal static class CommonInterest
             new SyncBlockingMethod(Namespaces.SystemThreadingTasks, nameof(Task), nameof(Task<int>.Result), null),
         };
 
+        /// <summary>
+        /// This is an explicit rule to ignore the code that was generated by Xaml2CS.
+        /// </summary>
+        /// <remarks>
+        /// The generated code has the comments like this:
+        /// <![CDATA[
+        ///   //------------------------------------------------------------------------------
+        ///   // <auto-generated>
+        /// ]]>
+        /// This rule is based on the fact the keyword "&lt;auto-generated&gt;" should be found in the comments.
+        /// </remarks>
+        internal static bool ShouldIgnoreContext(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = context.Node.FirstAncestorOrSelf<NamespaceDeclarationSyntax>();
+            if (namespaceDeclaration != null)
+            {
+                foreach (var trivia in namespaceDeclaration.NamespaceKeyword.GetAllTrivia())
+                {
+                    const string autoGeneratedKeyword = @"<auto-generated>";
+                    if (trivia.FullSpan.Length > autoGeneratedKeyword.Length
+                        && trivia.ToString().Contains(autoGeneratedKeyword))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        internal static void InspectMemberAccess(SyntaxNodeAnalysisContext context, MemberAccessExpressionSyntax memberAccessSyntax, DiagnosticDescriptor descriptor, IReadOnlyList<SyncBlockingMethod> problematicMethods)
+        {
+            if (descriptor == null)
```

Comment:
Sadly, no. Extra assembly dependencies that analyzers have greatly complicate the story.